Shared/Connect-M365.ps1

```diff
@@ -0,0 +1,159 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function Test-EXOConnection {
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    [OutputType([bool])]
+    param (
+        [Switch]$Force
+    )
+    #Validate EXO V2 is installed
+    if ((Get-Module -ListAvailable | Where-Object { $_.Name -like "ExchangeOnlineManagement" }).count -ge 1) {
+        Write-Host "ExchangeOnlineManagement Powershell Module installed"
+    } else {
+        if ($Force -or $PSCmdlet.ShouldContinue("Do you want to install the module?", "ExchangeOnlineManagement Powershell Module not installed")) {
+            Install-Module -Name ExchangeOnlineManagement -Force -ErrorAction SilentlyContinue -Scope CurrentUser
+            if ((Get-Module -ListAvailable | Where-Object { $_.Name -like "ExchangeOnlineManagement" }).count -ge 1) {
+                Write-Host "ExchangeOnlineManagement Powershell Module installed"
+            } else {
+                Write-Host "ExchangeOnlineManagement Powershell Module installation failed" -ForegroundColor Red
+                return $false
+            }
+        } else {
+            Write-Host "We cannot continue without ExchangeOnlineManagement Powershell module" -ForegroundColor Red
+            return $false
+        }
+    }
+
+    #Validate EXO V2 is loaded
+    if ((Get-Module | Where-Object { $_.Name -like "ExchangeOnlineManagement" }).count -ge 1) {
+        Write-Host "ExchangeOnlineManagement Powershell Module loaded"
+    } else {
+        Import-Module ExchangeOnlineManagement -ErrorAction SilentlyContinue -Force
+        if ((Get-Module | Where-Object { $_.Name -like "ExchangeOnlineManagement" }).count -ge 1) {
+            Write-Host "ExchangeOnlineManagement Powershell Module Imported"
+        } else {
+            Write-Host "ExchangeOnlineManagement Powershell Module Import failed" -ForegroundColor Red
+            return $false
+        }
+    }
+
+    #Validate EXO V2 is connected or try to connect
+    $connection = $null
+    $connection = Get-ConnectionInformation -ErrorAction SilentlyContinue
```

Comment:
`Get-ConnectionInformation` appears to be only for EXO v3. Might need to make sure we have that version. 

https://learn.microsoft.com/en-us/powershell/module/exchange/get-connectioninformation?view=exchange-ps